lib/utilities/request.js

```diff
@@ -0,0 +1,161 @@
+const url = require('url');
+const http = require('http');
+const https = require('https');
+const HttpProxyAgent = require('http-proxy-agent');
+const HttpsProxyAgent = require('https-proxy-agent');
+
+const { jsutil } = require('arsenal');
+const {
+    proxyCompareUrl,
+} = require('arsenal').storage.data.external.backendUtils;
+
+const validVerbs = ['HEAD', 'GET', 'POST', 'PUT', 'DELETE'];
+const updateVerbs = ['POST', 'PUT'];
+
+/*
+ * create a new header object from an existing header object. Similar keys
+ * will be ignored if a value has been set for theirlower-cased form
+ */
+function createHeaders(headers)  {
+    if (typeof headers !== 'object') {
+        return {};
+    }
+    const out = {};
+    Object.keys(headers).forEach(key => {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries
Object.entries would make this a little cleaner I think.